src/async.c

```diff
@@ -20,10 +16,12 @@ typedef struct {
 void xattr_get_execute(napi_env env, void* _data) {
   XattrGetData* data = _data;
 
-#ifdef __APPLE__
-  data->value_length = getxattr(data->filename, data->attribute, NULL, 0, 0, 0);
+#if BSD_OS == 1
+    data->value_length = extattr_get_file(data->filename, EXTATTR_NAMESPACE_USER, data->attribute, NULL, 0);
```

Comment:
Looking for all attributes in the `EXTATTR_NAMESPACE_USER` namespace is only valid if you're sure that the attribute name contains no `.` characters.  Are you sure that they won't?  Perhaps add an assertion to that effect.

Reply:
this code is fast version